modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -146,12 +148,21 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
             inp.push_back(inputs[0].size[i]);
             out.push_back(outputs[0].size[i]);
         }
-        if (globalPooling) {
-            kernel = Size(inp[1], inp[0]);
-            kernel_size = std::vector<size_t>(inp.begin(), inp.end());
+
+        if (kernel_size.size() > inp.size()) {
+            kernel_size.erase(kernel_size.begin());
         }
+        kernel_size.resize(out.size());
```

Comment:
I think we do not need this due in the next loop we can iterate over input shape and change `kernel_size`:

```cpp
        CV_Assert(isGlobalPooling.size() == kernel_size.size());
        std::vector<size_t> finalKernel;
        for (int i = 0; i < inp.size(); i++)
        {
            int idx = isGlobalPooling.size() - inp.size() + i;
            finalKernel.push_back(isGlobalPooling[idx] ? inp[i] : kernel_size[idx]);
        }
        kernel_size = finalKernel;
```

Reply:
Always  `isGlobalPooling.size() = 3`. If pooling type is `ROI` then kernel_size is empty.